schema/productconfig.rnc

```diff
@@ -0,0 +1,474 @@
+  }
+
+ds.type.xsltparam =
+  # An XSLT parameter name
+  xsd:token {
+    pattern="""[\-_\.\+a-zA-Z0-9]+"""
+  }
+
+ds.type.id =
+  # An ID within the document we are trying to build
+  xsd:token {
+    pattern="""[\-_\.\+a-zA-Z0-9]+"""
+  }
+
+
+# TAG/ATTRIBUTE SETS
+
+div
+{
+# FIXME: add img? (It is a bit unclear whether out script would need to take
+# care of copying stuff then.)
+ds.htmlblock =
+  ds.p
+  | ds.div
+  | ds.pre
+  | ds.ul
+  | ds.ol
+  | ds.h1
+  | ds.h2
+  | ds.h3
+  | ds.h4
+  | ds.h5
+  | ds.h6
+}
+
+div
+{
+ds.htmlinline =
+    ds.sup
+  | ds.u
+  | ds.cite
+  | ds.span
+  | ds.s
+  | ds.a
+  | ds.sub
+  | ds.strong
+  | ds.em
+  | ds.q
+  | ds.code
+}
+
+div
+{
+ds.htmlattr =
+  attribute id { xsd:ID }?,
+  attribute class { ds.type.class }?
+}
+
+
+
+# ROOT
+
+ds.productindex =
+  element productindex {
+    # This ID value can only be checked when we take all of the productconfig
+    # files together (i.e. after we have run stitch.sh). In any case, it does
+    # not make sense to use xsd:id here because there can only ever be one
+    # of these ID per productconfig file.
+    attribute productid { ds.type.id },
+    attribute lifecycle { "unpublished" | "beta" | "supported" | "unsupported" },
+    ds.product,
+    ds.docset+
+  }
+
+
+# PRODUCT DESCRIPTION
+
+ds.product =
+  element product {
+    ds.name,
+    ds.shortname?,
+    ds.maintainers,
+    ds.defaultblurb,
+    ds.blurb+
+  }
+
+ds.name =
+  element name {
+    text
+  }
+
+ds.shortname =
+  element shortname {
+    text
+  }
+
+ds.maintainers =
+  element maintainers {
+    ds.contact+
+  }
+
+ds.contact =
+  element contact {
+    ds.type.email
+  }
+
+ds.defaultblurb =
+  element defaultblurb {
+    attribute lang { ds.type.lang },
+    ds.htmlblock+
+  }
+
+ds.blurb =
+  element blurb {
+    attribute lang { ds.type.lang },
+    ds.htmlblock+
+  }
+
+
+# DOCSETS
+
+ds.docset =
+  element docset {
+    attribute setid { ds.type.id },
+    ds.version,
+    (
+      (ds.builddocs,
+       ds.extralinks?) |
+      ds.extralinks
+    )
+  }
+
+ds.version =
+  element version {
+    text
+  }
+
+ds.builddocs =
+  element builddocs {
+    ds.vcs,
+    ds.default,
+    ds.translation*
+  }
+
+ds.vcs =
+  element vcs {
+    # git is currently the only thing supported but let's leave that open
+    attribute type { "git" },
+    ds.remote,
+    ds.checkoutdir
+  }
+
+ds.remote =
+  element remote {
+    ds.type.gitremote
+  }
+
+ds.checkoutdir =
+  element checkoutdir {
+    ds.type.dirname
+  }
+
+ds.default =
+  element default {
+    attribute lang { ds.type.lang },
+    ds.branch,
+    ds.subdir?,
+    ds.deliverable+
+  }
+
+ds.branch =
+  element branch {
+    ds.type.branch
+  }
+
+ds.subdir =
+  element subdir {
+    ds.type.dirname
+  }
+
+ds.deliverable =
+  element deliverable {
+    # FIXME: implementation probably uses yes/no values, xsd:boolean supports 0/1 or true/false!
+    attribute remarks { xsd:boolean }?,
```

Comment:
Well, you could add this as a choice:

```
ds.type.boolean =
   xsd:boolean |  token "yes" | token "no"

# ...
  attribute remarks { ds.type.boolean }?,
```

Reply:
Nah, allowing 0/false/no makes no real sense to me and only makes this harder to implement.